# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/webidl/NavigationPreloadManager.webidl
# Commit: 40c22bd043ba
# Full Hash: 40c22bd043bac8fa0c7ee3618572b6b9f3769a76
# Author: Peter Van der Beken <peterv@propagandism.org>
# Date: 2022-05-03 15:57:51
# Description:
#   Bug 1764935 - Deal properly with Promise creation failure in NavigationPreloadManager. r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D144341
# ==============================================================================

diff -r 3f83c6e28484 -r 40c22bd043ba dom/webidl/NavigationPreloadManager.webidl
--- a/dom/webidl/NavigationPreloadManager.webidl	Tue May 03 08:11:27 2022 +0000
+++ b/dom/webidl/NavigationPreloadManager.webidl	Tue May 03 08:23:25 2022 +0000
@@ -10,9 +10,13 @@
 [Pref="dom.serviceWorkers.navigationPreload.enabled", SecureContext,
  Exposed=(Window,Worker)]
 interface NavigationPreloadManager {
+  [NewObject]
   Promise<void> enable();
+  [NewObject]
   Promise<void> disable();
+  [NewObject]
   Promise<void> setHeaderValue(ByteString value);
+  [NewObject]
   Promise<NavigationPreloadState> getState();
 };
 
